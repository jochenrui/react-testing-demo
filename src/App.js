function App() {
  return (
    <div className="App">
      <h1>React Testing</h1>
    </div>
  );
}

export default App;
