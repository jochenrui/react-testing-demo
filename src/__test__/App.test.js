import React from "react";

/**
 * Enzyme imports
 */
import Enzyme, { shallow, render, mount } from "enzyme";
// toJson -> convert shallow renders into JSON for Snapshot Tests
import toJson from "enzyme-to-json";
/**
 * Adapter -> makes enzyme work with specific React version ->
 * takes care of changes in between React versions, so that Enzyme can stay unchanged
 */
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

/**
 * react-test-renderer
 */
import * as renderer from "react-test-renderer";

import App from "../App";

Enzyme.configure({ adapter: new Adapter() });

// snapshot test
describe("app renders correctly", () => {
  // shallow -> renders high level components, no children
  test("app shallow renders correctly", () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // mount -> renders full tree
  test("app mount renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
