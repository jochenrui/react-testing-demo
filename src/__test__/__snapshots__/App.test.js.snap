// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly app mount renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1>
      React Testing
    </h1>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-around",
      }
    }
  >
    <button
      role="button-role"
      style={
        Object {
          "borderRadius": "7px",
          "height": "40px",
          "width": "100px",
        }
      }
    >
      Role
    </button>
    <button
      data-testid="button-testid"
      style={
        Object {
          "borderRadius": "7px",
          "height": "40px",
          "width": "100px",
        }
      }
    >
      Test Id
    </button>
    <button
      className="button-class"
      style={
        Object {
          "borderRadius": "7px",
          "height": "40px",
          "width": "100px",
        }
      }
    >
      Test Id
    </button>
  </div>
</div>
`;

exports[`app renders correctly app shallow renders correctly 1`] = `
<div>
  <Heading />
  <IdentifierButtons />
</div>
`;
